### ISSUE TITLE
Consider upgrade to Hibernate ORM 5.2

### ISSUE THREAD

**Issue by Polve** [spring-projects/spring-boot#6111]
Hibernate 5.2 is now GA and it might be nice to upgrade as part of Spring Boot 1.4.

**Comment by philwebb** [comment]
A lot of things have changed between 5.1 and 5.2 (Java 8 minimum requirement being the most concerning). Ideally it would be good to get this into Spring Boot 1.4 as people will be upgrading from Hibernate 4.x and it would be nice to get all the pain out of the way in one hit.

**Comment by rakpan** [comment]
Are we raising the bar for spring boot too?

**Comment by philwebb** [comment]
@rakpan Not until Spring Boot 2.0. However, we could still consider Hibernate 5.2 as our default choice as long as people can easily downgrade.

**Comment by snicoll** [comment]
Considering that `hibernate-entitymanager` is gone in 5.2, switching to 5.2 effectively means that 5.2 is our bare minimum since it changes the pom structure. I'd argue `5.2.1` should restore that artifact (even if it's empty) for backward compatible reason.

@philwebb I had a quick look and the "as long as people can easily downgrade" does not seem possible at the moment.

**Comment by philwebb** [comment]
@snicoll Unless I'm missing something, I'm not sure I agree. Users might not be able to downgrade with a single property but they could set the property _and_ add back `hibernate-entitymanager`.

**Comment by snicoll** [comment]
Yes. I don't consider that as "easy".

**Comment by philwebb** [comment]
@snicoll It's not super easy, but it's not that hard and it's probably less confusing that having an empty `hibernate-entitymanager` project that doesn't contain anything.

**Comment by gniding** [comment]
My approach to use hibernate 5.2 right now would be  

		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-data-jpa</artifactId>
			<exclusions>
				<exclusion>
					<groupId>org.hibernate</groupId>
					<artifactId>hibernate-entitymanager</artifactId>
				</exclusion>
			</exclusions>
		</dependency>
		<dependency>

**Comment by wilkinsona** [comment]
Spring Framework 4.3.0.RELEASE isn't fully compatible with Hibernate 5.1 (see [SPR-14365]([https://jira.spring.io/browse/SPR-14365) so we either need to move to 5.2 or wait for Spring Framework 4.3.1.RELEASE.

**Comment by snicoll** [comment]
Indeed, that probably means that Our RC1 has to wait `4.3.1.RELEASE`

**Comment by snicoll** [comment]
Fixed in 2ff9e3c

**Comment by ceefour** [comment]
I'd :+1: not for "requiring" 5.2 but at least allowing people to use Hibernate 5.2 if they want.
Currently when I used hibernate-core and hibernate-spatial 5.2.1.Final, with Spring Boot 1.4.0, I got:

```
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'serverProperties' defined in class path resource [org/springframework/boot/autoconfigure/web/ServerPropertiesAutoConfiguration.class]: Initialization of bean failed; nested exception is java.lang.NoClassDefFoundError: org/hibernate/bytecode/instrumentation/internal/FieldInterceptionHelper
```

While I'm ok with Spring Boot not defaulting to Hibernate 5.2, it concerns me that users can't use Hibernate 5.2 or even 5.3.

**Comment by snicoll** [comment]
@ceefour this issue is closed. If you believe you have found an issue in Spring Boot, please create a dedicated issue. Before you do, try your app without `hibernate-spatial`. Maybe that release hasn't been aligned with Hibernate `4.2.1.Final` yet? We have a sample that works with Hibernate 4.2.

**Comment by ceefour** [comment]
@snicoll We have no issue with Hibernate 4.2 and Hibernate 5.0 (both Spring Boot 1.3.6 and 1.4.0), just Boot 1.4.0 (Spring v4.3.2.RELEASE) with Hibernate 5.2.1.Final.

Removing hibernate-special doesn't remove the problem.

Should I file the issue on Spring-boot or JIRA Spring ORM ?

Spring 4.3 claims it supports Hibernate 4.2 (https://jira.spring.io/browse/SPR-14327) so I'm not sure whether this is Boot's or Spring's...

**Comment by snicoll** [comment]
Sorry, I mixed up. I meant `5.2.1.Final`. We have a [sample](https://github.com/spring-projects/spring-boot/tree/master/spring-boot-samples/spring-boot-sample-hibernate52) so I don't know where the problem is but it might be in your app.

**Comment by ceefour** [comment]
Thanks @snicoll , I've found the issue. It's had I still loaded `hibernate-entitymanager 5.0.9.Final`, which to fix it in my project requires:

```
        <dependency>
            <groupId>org.jadira.usertype</groupId>
            <artifactId>usertype.core</artifactId>
            <version>5.0.0.GA</version>
            <!-- usertype 5.0.0 not yet compatible with Hibernate 5.2: https://github.com/JadiraOrg/jadira/pull/57 -->
            <exclusions>
                <exclusion>
                    <groupId>org.hibernate</groupId>
                    <artifactId>hibernate-entitymanager</artifactId>
                </exclusion>
            </exclusions>
        </dependency>

        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-data-jpa</artifactId>
            <!-- hibernate-entitymanager must be excluded: https://github.com/spring-projects/spring-boot/issues/6111#issuecomment-225074425 -->
            <exclusions>
                <exclusion>
                    <groupId>org.hibernate</groupId>
                    <artifactId>hibernate-entitymanager</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
```

However, usertype's compatibility with Hibernate 5.2 is another issue (https://github.com/JadiraOrg/jadira/pull/57).

It's nice that the Spring Boot documentation has been updated to refer to the Hibernate 5.2 sample, however I'd say it's a good idea to signify in the documentation that especially the exclusion of hibernate-entitymanager is strictly required. The Hibernate 5.2 sample's pom.xml has no information at all on why hibernate-entitymanager was excluded. So users can be frustrated for some time before finding this page which explains the background of issue.

**Comment by snicoll** [comment]
@ceefour could you please stop commenting on closed issues please?

**Comment by ceefour** [comment]
@snicoll Sorry, this would be my last comment on this thread. I did the comment because this page is high on Google, so if someone else is having the same problem as I am, I hope that by commenting, it can lead them to the solution (instead of being even more confused, as it was in my case). It happens a lot to me, and I'm thankful for people "leaving a trail" of problem solving. I know this is not StackOverflow though, I'm sorry.

**Comment by re-pvsr** [comment]
Sorry to comment on a closed issue, but @snicoll gave this link [https://github.com/spring-projects/spring-boot/tree/master/spring-boot-samples/spring-boot-sample-hibernate52](url) for a sample pom to upgrade to 5.2. The link is dead. How do we upgrade to 5.2, do we just change the hibernate.version to 5.2.1 and that's it ? I am using 1.5.7 Boot

**Comment by re-pvsr** [comment]
ok, got it - This is the new link https://github.com/spring-projects/spring-boot/tree/v1.5.6.RELEASE/spring-boot-samples/spring-boot-sample-hibernate52

Sorry, this issue link is coming up first in the google search results and hope this can help someone.
